### ISSUE TITLE
Allow me to filter by branch

### ISSUE THREAD

**Issue by OsirisTerje** [ALM-Rangers/Visualize-Team-Project-Health-Widgets#8]
When monitoring I only want to see builds that are triggered on the master branch (or any specified branch), even if the same build can be used for multiple branches.

**Comment by WouterDeKort** [comment]
Hi @OsirisTerje,

thanks for the suggestion. I'll add it to the backlog. I don't think it's really hard to implement. Would a dropdown for the branch be OK or do you scenarios where you want to select multiple branches?

Thanks,
Wouter

**Comment by OsirisTerje** [comment]
As far as I can see,  one should be enough, so a dropdown would be fine

**Comment by KenticoMartinS** [comment]
Hi,
I want to ask what's the state of this feature? Is it planned or in progress? We are really looking forward to this feature. We are building all commits in all branches, but on the dashboard, we are only interested in the result of the master branch.

**Comment by mattwhitfield** [comment]
Bump, same again. Filtering the dashboard to master is what we are interested in. If there's a red build in a feature branch - we don't really care. Red build in a master branch is a massive issue and we need to jump on it straight away.

**Comment by sanderaernouts** [comment]
@mattwhitfield @KenticoMartinS if you are looking specifically for the status of a build, the new build widget in Azure DevOps does support filtering on branches:
![image](https://user-images.githubusercontent.com/6184673/47156146-c12c0a00-d2e6-11e8-8d32-09ef2de4f3e1.png)

**Comment by mattwhitfield** [comment]
@sanderaernouts Yes - we do use that today. It was the 'Build Health Details' widget I meant specifically. That allows you to select the build definition / pipeline - but doesn't have a branch filter...

**Comment by WouterDeKort** [comment]
Sorry people. As you know this is an open source project I run in the evenings and I haven't found any time lately. 

I'll try to schedule something for the next couple of weeks. If one of you wants to do a pull request that would be great and I would love to help you get it set up.

Wouter

**Comment by mattwhitfield** [comment]
@WouterDeKort - certainly nothing to apologise for. I know how life gets in the way. I'd be up for sorting this out at some point - so if you do have some pointers on how to get started with that then I could probably knock that out at some point :)

**Comment by WouterDeKort** [comment]
You can develop the extension locally with VS Code , your own Azure DevOps organization and a Marketplace account. The code is in Typescript and some HTML and CSS. After your pull request a Azure DevOps pipeline runs some checks and deploys to the Marketplace. 

Let me know if got have any questions. 

Wouter

**Comment by jakobehn** [comment]
I should have some time next week to fix this one, we’re using it at my current client.

**Comment by WouterDeKort** [comment]
@jakobehn Maybe make it optional? If it's a simple text field people can decide for themselves?

**Comment by mattwhitfield** [comment]
@jakobehn Personally I am the same as you - the branch is obvious from the build number because that makes it easier in a bunch of other places (e.g. build statuses posted to slack). Many many thanks for addressing this.

**Comment by OsirisTerje** [comment]
@jakobehn  @WouterDeKort 's suggestion is good, because you don't know if people will have that convention or not.   By default I would like to see the branch name explicitly, if it doesn't get too crowded.  Do you have an example screenshot with or without ?

**Comment by Tyler-Keith-Thompson** [comment]
What will it take to make this happen? Looks like the work was done a while ago and it's just waiting for a review? Our team could really use this feature.

**Comment by li-dawei** [comment]
Another upvote for above feature, this would be really useful for us too, any chance of releasing this soon?
